SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialTraceEventBuilder.h

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Interop/Connection/SpatialTraceEvent.h"
+#include "SpatialCommonTypes.h"
+#include "WorkerSDK/improbable/c_worker.h"
+
+#define GDK_EVENT_NAMESPACE "unreal_gdk."
+
+namespace SpatialGDK
+{
+class FSpatialTraceEventBuilder
+{
+public:
+	FSpatialTraceEventBuilder(const char* InType);
+	FSpatialTraceEventBuilder(const char* InType, const FString& InMessage);
+
+	// --- Builder Functions ---
+
+	FSpatialTraceEventBuilder AddObject(const UObject* Object);
+	FSpatialTraceEventBuilder AddObject(const FString& Key, const UObject* Object);
```

Comment:
I think we should be passing keys into all of these rather than doing it implicitly. You don't actually have to call these functions in many places so I think the less implicit stuff there is the better.